src/dodal/common/types.py

```diff
@@ -18,4 +18,4 @@
 
 class UpdatingDirectoryProvider(DirectoryProvider, ABC):
     @abstractmethod
-    def update(self, **kwargs) -> None: ...
```

Comment:
This change is required for Hyperion, please revert (or rebase on top of master)